scripts/py2md.py

```diff
@@ -223,10 +230,56 @@ def main():
             render_module_header=False,
         ),
     )
+    if out:
+        out_path = Path(out)
+        os.makedirs(out_path.parent, exist_ok=True)
 
     pydocmd.load_modules()
     pydocmd.process()
     pydocmd.render()
+    logging.info("Processed %s", module)
+
+
+def _py2md_wrapper(x: Tuple[str, str]):
+    """
+    Used to wrap py2md since we can't pickle a lambda (needed for multiprocessing).
```

Comment:
I wrote a library called "bettermap" at https://github.com/allenai/bettermap.

The gist of it is that normal Python multiprocessing foregoes some optimization and convenience when it comes to multiprocessing, and it does this to be compatible with Windows. If you don't care about Windows, you can just fork the process. That's what `bettermap` does.

You would say `bettermap.map_per_process(fn, sequence)`, and it takes care of the rest. `fn` can be any callable, and your Python source file doesn't have to be importable. You can do this from a console if you want. Due to the magic of forking, the elements of `sequence` don't have to be serialized (or even serializable).

So it has a bunch of advantages. On the other hand, it's an extra dependency (or you have to copy-and-paste the code in here). I'll leave it to your judgement whether you want to use it.

Reply:
This is neat! Unfortunately on my first try using it, it turned out to be slower here because the `py2md()` function has to do some module importing behind the scenes. With `multiprocessing.Pool` those imports only have to happen once per thread, but with `better.map_per_process` they are imported every time the `py2md()` function is called.

I'm guessing there's a way around this. Maybe by importing everything in the main process before `map_per_process` is called?

Reply:
Yes, that would do it.